### DISCUSSION TITLE
No communication with Webinterface

### DISCUSSION THREAD

**Discussion by NitramH** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#159] [Q&A]
Hi,

first of all, thanks for this project, great work!

I commissioned my device with Miami 2021. Startup is fine, initialisation works fine.
System is reacting on button interaction at the Miami panel.

But the web interface refuses communication.
There is no interaction to and from the ESP possible.
Temp = 0, Target temp = 0 ...., no reaction on switches ...
Text "Spa Control" goes into red color.

Does anyone has an proposal where and what to search for?

Remark: My Miami device is during the commissioning of the ESP disconnected from the rest of the pool (dry).
Could this be a reason? Sure, I do not use the heating or pump function not to damage the components.
But web interface acts independent from this, I guess.

Thanks in advance,
Martin

P.S.:
I´m new to Visualstudio, what is the difference between "General/Upload" and "Platform/Upload Filesystem"?

**Comment by visualapproach** [comment]
Hi! 
> I´m new to Visualstudio, what is the difference between "General/Upload" and "Platform/Upload Filesystem"?

The flash memory on the ESP is divided between program memory and a file system. You can think of it as ROM + HDD. 
General/upload uploads the program and Platform/upload uploads all the data files, like "index.html". You must first build filesystem image before uploading, i think.

> ext "Spa Control" goes into red color.

This is because the ESP has disconnected. I can think of two possible reasons.
1 - ESP entered the auto portal. Check if there is a wifi access point called "auto portal".
2 - ESP crashed or reboots. Insufficient power has caused this on 4-wire systems, but since you didn't mention E13 I reckon you have a 6-wire system. (Do not try to power with USB AND pump at the same time)
Not sure it will help, but you could look in the browser's "console" to see what messages are sent.

**Reply by NitramH** [reply]
Thank you for your great and fast support!

I think it is 2. I use very thin data wires in a shielded data cable and didn´t thought about currents and voltage drop at 5V and GND.
Will check, how I can get the messages and will reply later today.

**Comment by NitramH** [comment]
I got it running on Chrome browser at the Laptop without problems. Console shows expected status messages from the ESP in short frequency (ca. 1Hz). Then I returned to mobile browser.
Finally root cause is my WiFi environment setup and not the ESP. I have two access points with different names. Seems that communication between them (ESP at the first, mobile phone at the second) gets interrupted. I opened it in the same WiFi environment and it works now fine.

Next step is to enable the MQTT and integrate it into Home Assistant and then enjoying a bath in the Miami Spa at new years eve :-D

P.S.: I have 3 spare PCBs, anyone in Europe interested in? (made for 6-wire solution, ESP D1 mini and two level shifter setup)


![Unbenannt](https://user-images.githubusercontent.com/96552693/147229514-0e31acf5-525b-4926-a1cb-4a14eeef45ad.JPG)

**Reply by visualapproach** [reply]
Nice! There are some examples in this git that should get you going with HA integration.
May I suggest you post the PCB offering in the relevant discussion topic also.
